compose/material/material-navigation/src/commonMain/kotlin/androidx/compose/material/navigation/BottomSheetNavigator.kt

```diff
@@ -0,0 +1,130 @@
+     */
+    public val currentValue: ModalBottomSheetValue
+        get() = sheetState.currentValue
+
+    /**
+     * @see ModalBottomSheetState.targetValue
+     */
+    public val targetValue: ModalBottomSheetValue
+        get() = sheetState.targetValue
+}
+
+/**
+ * Create and remember a [BottomSheetNavigator]
+ */
+@Composable
+public fun rememberBottomSheetNavigator(
+    animationSpec: AnimationSpec<Float> = SpringSpec()
+): BottomSheetNavigator {
+    val sheetState = rememberModalBottomSheetState(
+        ModalBottomSheetValue.Hidden,
+        animationSpec = animationSpec
+    )
+    return remember(sheetState) { BottomSheetNavigator(sheetState) }
+}
+
+/**
+ * Navigator that drives a [ModalBottomSheetState] for use of [ModalBottomSheetLayout]s
+ * with the navigation library. Every destination using this Navigator must set a valid
+ * [Composable] by setting it directly on an instantiated [Destination] or calling
+ * [androidx.compose.material.navigation.bottomSheet].
+ *
+ * <b>The [sheetContent] [Composable] will always host the latest entry of the back stack. When
+ * navigating from a [BottomSheetNavigator.Destination] to another
+ * [BottomSheetNavigator.Destination], the content of the sheet will be replaced instead of a
+ * new bottom sheet being shown.</b>
+ *
+ * When the sheet is dismissed by the user, the [state]'s [NavigatorState.backStack] will be popped.
+ *
+ * The primary constructor is not intended for public use. Please refer to
+ * [rememberBottomSheetNavigator] instead.
+ *
+ * @param sheetState The [ModalBottomSheetState] that the [BottomSheetNavigator] will use to
+ * drive the sheet state
+ */
+public expect class BottomSheetNavigator(
+    sheetState: ModalBottomSheetState
+) : Navigator<Destination> {
+
+    internal val sheetState: ModalBottomSheetState
+
+    /**
+     * Get the transitionsInProgress from the [state]. In some cases, the [sheetContent] might be
+     * composed before the Navigator is attached, so we specifically return an empty flow if we
+     * aren't attached yet.
+     */
+    internal val transitionsInProgress: StateFlow<Set<NavBackStackEntry>>
+
+    /**
+     * Access properties of the [ModalBottomSheetLayout]'s [ModalBottomSheetState]
+     */
+    public val navigatorSheetState: BottomSheetNavigatorSheetState
+
+    /**
+     * A [Composable] function that hosts the current sheet content. This should be set as
+     * sheetContent of your [ModalBottomSheetLayout].
+     */
+    internal val sheetContent: @Composable ColumnScope.() -> Unit
+
+    /**
+     * [NavDestination] specific to [BottomSheetNavigator]
+     */
+    public class Destination(
+        navigator: BottomSheetNavigator,
+        content: @Composable ColumnScope.(NavBackStackEntry) -> Unit
+    ) : NavDestination, FloatingWindow {
+        internal val content: @Composable ColumnScope.(NavBackStackEntry) -> Unit
+    }
+
+    internal companion object {
+        internal val NAME: String
```

Comment:
We don't need this in "expect" part and on Android

Reply:
done. thanks